x-pack/plugins/apm/public/components/shared/time_comparison/index.test.tsx

```diff
@@ -0,0 +1,142 @@
+import { IUrlParams } from '../../../context/url_params_context/types';
+import {
+  expectTextsInDocument,
+  expectTextsNotInDocument,
+} from '../../../utils/testHelpers';
+import { TimeComparison } from './';
+import * as urlHelpers from '../../shared/Links/url_helpers';
+
+function getWrapper(params?: IUrlParams) {
+  return ({ children }: { children?: ReactNode }) => {
+    return (
+      <MemoryRouter>
+        <MockUrlParamsContextProvider params={params}>
+          <EuiThemeProvider>{children}</EuiThemeProvider>
+        </MockUrlParamsContextProvider>
+      </MemoryRouter>
+    );
+  };
+}
+
+describe('TimeComparison', () => {
+  const spy = jest.spyOn(urlHelpers, 'replace');
+  beforeEach(() => {
+    jest.resetAllMocks();
+  });
+  describe('Time range is between 0 - 24 hours', () => {
+    it('sets default values', () => {
+      const Wrapper = getWrapper({
+        start: '2021-01-28T14:45:00.000Z',
+        end: '2021-01-28T15:00:00.000Z',
+      });
+      render(<TimeComparison />, {
+        wrapper: Wrapper,
+      });
+      expect(spy).toHaveBeenCalledWith(expect.anything(), {
+        query: {
+          comparisonEnabled: 'true',
+          comparisonType: 'yesterday',
+        },
+      });
+    });
+    it('selects yesterday and enables comparison', () => {
+      const Wrapper = getWrapper({
+        start: '2021-01-28T14:45:00.000Z',
+        end: '2021-01-28T15:00:00.000Z',
+        comparisonEnabled: true,
+        comparisonType: 'yesterday',
+      });
+      const component = render(<TimeComparison />, {
+        wrapper: Wrapper,
+      });
+      expectTextsInDocument(component, ['Yesterday', 'A week ago']);
+      expect(
+        (component.getByTestId('comparisonSelect') as HTMLSelectElement)
+          .selectedIndex
+      ).toEqual(0);
+    });
+  });
+
+  describe('Time range is between 24 hours - 1 week', () => {
+    it('sets default values', () => {
+      const Wrapper = getWrapper({
+        start: '2021-01-26T15:00:00.000Z',
+        end: '2021-01-28T15:00:00.000Z',
+      });
+      render(<TimeComparison />, {
+        wrapper: Wrapper,
+      });
+      expect(spy).toHaveBeenCalledWith(expect.anything(), {
+        query: {
+          comparisonEnabled: 'true',
+          comparisonType: 'week',
+        },
+      });
+    });
+    it('selects week and enables comparison', () => {
+      const Wrapper = getWrapper({
+        start: '2021-01-26T15:00:00.000Z',
+        end: '2021-01-28T15:00:00.000Z',
+        comparisonEnabled: true,
+        comparisonType: 'week',
+      });
+      const component = render(<TimeComparison />, {
+        wrapper: Wrapper,
+      });
+      expectTextsNotInDocument(component, ['Yesterday']);
+      expectTextsInDocument(component, ['A week ago']);
+      expect(
+        (component.getByTestId('comparisonSelect') as HTMLSelectElement)
+          .selectedIndex
+      ).toEqual(0);
+    });
+  });
+
+  describe('Time range is greater than 8 days', () => {
```

Comment:
Should be 7 days, right?

```suggestion
  describe('Time range is greater than 7 days', () => {
```

Reply:
yes thanks!